internal/cloud/realm/logs.go

```diff
@@ -0,0 +1,121 @@
+package realm
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+const (
+	logsPathPattern = appPathPattern + "/logs"
+
+	logsQueryEndDate    = "end_date"
+	logsQueryErrorsOnly = "errors_only"
+	logsQueryStartDate  = "start_date"
+	logsQueryType       = "type"
+
+	logsDateFormat = "2006-01-02T15:04:05.999Z07:00"
+)
+
+// set of supported Realm app log types
+const (
+	LogTypeAPI                   = "API"
+	LogTypeAPIKey                = "API_KEY"
+	LogTypeAuth                  = "AUTH"
+	LogTypeAuthTrigger           = "AUTH_TRIGGER"
+	LogTypeDBTrigger             = "DB_TRIGGER"
+	LogTypeFunction              = "FUNCTION"
+	LogTypeGraphQL               = "GRAPHQL"
+	LogTypePush                  = "PUSH"
+	LogTypeScheduledTrigger      = "SCHEDULED_TRIGGER"
+	LogTypeSchemaAdditiveChange  = "SCHEMA_ADDITIVE_CHANGE"
+	LogTypeSchemaGeneration      = "SCHEMA_GENERATION"
+	LogTypeSchemaValidation      = "SCHEMA_VALIDATION"
+	LogTypeServiceFunction       = "SERVICE_FUNCTION"
+	LogTypeServiceStreamFunction = "SERVICE_STREAM_FUNCTION"
+	LogTypeStreamFunction        = "STREAM_FUNCTION"
+	LogTypeSyncClientWrite       = "SYNC_CLIENT_WRITE"
+	LogTypeSyncConnectionEnd     = "SYNC_CONNECTION_END"
+	LogTypeSyncConnectionStart   = "SYNC_CONNECTION_START"
+	LogTypeSyncError             = "SYNC_ERROR"
+	LogTypeSyncOther             = "SYNC_OTHER"
+	LogTypeSyncSessionEnd        = "SYNC_SESSION_END"
+	LogTypeSyncSessionStart      = "SYNC_SESSION_START"
+	LogTypeWebhook               = "WEBHOOK"
+)
```

Comment:
is this necessary? it would be really nice to be able to control this from baas since logging is more fluid in nature of adding new types.

Reply:
based on the way our logs api works currently, yes.  the directive from product here has been "keep it like the UI" and for better or worse, a lot of this logic is pushed to the frontend (so we're more or less duplicating it here)

ive communicated this to drew and sumedha that should we ever make changes to log types and/or names, we have multiple places to remember to update.  hopefully this is something we can address with the logs rework

Reply:
I'm fine doing it this way for now, but this will add to the logs project scope coming up.